File: packages/dataframe/rust/stats/distributions/wilcoxon.test.R
#!/usr/bin/env Rscript

# Wilcoxon distribution test script
# Usage: Rscript wilcoxon.test.R <function> <x_or_p> <m> <n> [lower_tail] [log_p]

source("test-helpers.R")

# Define Wilcoxon distribution functions
wilcoxon_density <- function(x, m, n, log = FALSE) {
  dwilcox(x, m, n, log = log)
}

wilcoxon_cumulative <- function(x, m, n, lower.tail = TRUE, log.p = FALSE) {
  pwilcox(x, m, n, lower.tail = lower.tail, log.p = log.p)
}

wilcoxon_quantile <- function(p, m, n, lower.tail = TRUE, log.p = FALSE) {
  qwilcox(p, m, n, lower.tail = lower.tail, log.p = log.p)
}

wilcoxon_random <- function(nn, m, n) {
  rwilcox(nn, m, n)
}

# Run the test
run_distribution_test(wilcoxon_density, wilcoxon_cumulative, wilcoxon_quantile, wilcoxon_random)

File: packages/dataframe/rust/stats/distributions/wilcoxon.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust Wilcoxon functions...\n");

  const testCases = [
    { func: "dwilcox", distribution: "wilcoxon", args: [6, 3, 4, 0] },
    { func: "pwilcox", distribution: "wilcoxon", args: [6, 3, 4, 1, 0] },
    {
      func: "qwilcox",
      distribution: "wilcoxon",
      args: [0.5714286, 3, 4, 1, 0],
    },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test rwilcox separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const rwilcoxResult = await callR("rwilcox", "wilcoxon", 1, 3, 4);
    console.log(`rwilcox(3, 4) = ${rwilcoxResult}`);
  } catch (error) {
    console.log(`rwilcox: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
